src/harness/externalCompileRunner.ts

```diff
@@ -1,14 +1,17 @@
 /// <reference path="harness.ts"/>
 /// <reference path="runnerbase.ts" />
+const fs = require("fs");
+const path = require("path");
 abstract class ExternalCompileRunnerBase extends RunnerBase {
     abstract testDir: string;
-    public enumerateTestFiles() {
+    abstract report(result: any, cwd: string): string;

```

Comment:
Since they're all that's used, I'd type `result` as `{status: number, stdout: Buffer, stderr: Buffer}` here. Probably in the form of an interface so it can also be referenced in the implementations.